### PR TITLE
Functionality code ready

### PR DESCRIPTION
**Pull request by creativeappdevamit**
# What and why?

# How to test?

# Trello
https://trello.com/c/680HdmWq/778-s40-user-account-creation-sharing-and-transferring-rops-alterations

### CODE REVIEW

File: response_operations_ui/views/respondents.py [hunk 1]
```diff
@@ -110,12 +154,106 @@ def respondent_details(respondent_id):
...
         enrolments=enrolments,
         breadcrumbs=breadcrumbs,
         mark_for_deletion=account["mark_for_deletion"],
+        pending_transfer_surveys=formatted_transfer_surveys,
+        pending_share_surveys=formatted_share_surveys,
+        respondent_id=respondent_id,
+    )
+
+
+def get_formatted_pending_surveys(pending_surveys: list):
```
Comment (insacuri) [inline response_operations_ui/views/respondents.py:181]: Minor comment, you can change this to
```python
def get_formatted_pending_surveys(pending_surveys: list) -> list:
```
and then you can remove the rtype and type from the docstring for this function

File: response_operations_ui/views/respondents.py [hunk 2]
```diff
@@ -110,12 +154,106 @@ def respondent_details(respondent_id):
...
+                    "batch_no": batch_number,
+                    "recipient_email": recipient_email,
+                    "time_shared": convert_events_to_new_format(time_shared),
+                    "pending_survey_details": business_surveys_list,
+                }
+            )
+    return formatted_pending_surveys
+
+
+def convert_events_to_new_format(date):
```
Comment (insacuri) [inline response_operations_ui/views/respondents.py:220]: Can you do type hints for this function?  I assume date is an str, and the return type is also a string?  That's just a guess

File: response_operations_ui/views/respondents.py [hunk 3]
```diff
@@ -110,12 +154,106 @@ def respondent_details(respondent_id):
...
+    :return: formatted date
+    """
+    try:
+        date_time = parse_date(date)
+    except ParseError:
+        raise ParseError
+    return ordinal_date_formatter("{S} %B %Y at %H:%M", date_time)
+
+
+def suffix(day: int):
```
Comment (insacuri) [inline response_operations_ui/views/respondents.py:234]: Return type for this please (str)

File: response_operations_ui/views/respondents.py [hunk 4]
```diff
@@ -110,12 +154,106 @@ def respondent_details(respondent_id):
...
+    """
+    This function creates the ordinal suffix
+
+    :param: day of the date time object
+    :return: ordinal suffix
+    """
+    return "th" if 11 <= day <= 13 else {1: "st", 2: "nd", 3: "rd"}.get(day % 10, "th")
+
+
+def ordinal_date_formatter(date_format_required: str, date_to_be_formatted: datetime):
```
Comment (insacuri) [inline response_operations_ui/views/respondents.py:244]: Return typehint here as well please

**Review by insacuri — CHANGES_REQUESTED** [review]
A few small changes but looks good

File: response_operations_ui/views/respondents.py (hunk 1 above)
Comment (creativeappdevamit) [reply response_operations_ui/views/respondents.py:181]: ok

File: response_operations_ui/views/respondents.py (hunk 2 above)
Comment (creativeappdevamit) [reply response_operations_ui/views/respondents.py:220]: ok

**Review by insacuri — APPROVED** [review]

**Pull request merged**